### PR TITLE
Convert Remote Access CFN Template from JSON to YAML

### PR DESCRIPTION
**Pull request by allez-allez-allez**
Converted to YAML for better readability.

### CODE REVIEW

**Comment by userhas404d** [comment]
Looks like travis isn't happy :( The yaml lint outputs in the travis run results should be pretty straight forward (you can add the vscode plugin to your local install or run that check via make on your box locally prior to commit).  You'll also need to update this block in the Makefile to run separate version checks against json and yaml: https://github.com/plus3it/cfn/blob/master/Makefile#L53

File: templates/ra_guac_autoscale_public_alb.template.cfn.yaml [hunk 1]
```diff
@@ -0,0 +1,623 @@
...
+    'Fn::Transform':
+      Name: 'AWS::Include'
+      Parameters:
+        Location: 's3://app-chemistry/snippets/instance_type_map.snippet.cfn.yaml'
+Metadata:
+  Version: 0.3.1
+  cfn-lint:
+    config:
+      ignore_checks:
+        - E7001
```
Comment (userhas404d) [inline templates/ra_guac_autoscale_public_alb.template.cfn.yaml]: just curious, but does cfn-lint still alert on these after the conversion?

File: templates/ra_guac_autoscale_public_alb.template.cfn.yaml [hunk 2]
```diff
@@ -0,0 +1,623 @@
...
+  TargetGroupName:
+    Description: Name of the ALB Target Group
+    Value: !Ref ALBTargetGroup
+Parameters:
+  AmiId:
+    Default: ''
+    Description: (Optional) AMI ID -- will supersede Lambda-based AMI lookup using AmiNameSearchString
+    Type: String
+  AmiNameSearchString:
+    Default: amzn-ami-hvm-2017.09*.2017*-x86_64-gp2
```
Comment (userhas404d) [inline templates/ra_guac_autoscale_public_alb.template.cfn.yaml]: I get that this is just a default, but we should probably update it to a current year

File: templates/ra_guac_autoscale_public_alb.template.cfn.yaml [hunk 3]
```diff
@@ -0,0 +1,623 @@
...
+      - c5.xlarge
+      - c5.2xlarge
+      - c5.4xlarge
+      - m5.large
+      - m5.xlarge
+      - m5.2xlarge
+      - m5.4xlarge
+      - r4.large
+      - r4.xlarge
+    Default: t2.micro
```
Comment (userhas404d) [inline templates/ra_guac_autoscale_public_alb.template.cfn.yaml]: should probably bump this to a t3.micro

File: templates/ra_guac_autoscale_public_alb.template.cfn.yaml [hunk 4]
```diff
@@ -0,0 +1,623 @@
...
+  LdapServer:
+    Default: ad.example.com
+    Description: Name of LDAP server Guacamole will authenticate against
+    MinLength: '1'
+    Type: String
+  MaxCapacity:
+    Default: '2'
+    Description: The maximum number of instances for the autoscale group
+    MinLength: '1'
+    Type: String
```
Comment (userhas404d) [inline templates/ra_guac_autoscale_public_alb.template.cfn.yaml]: more of a nit but this should probably be the `Number` type

File: templates/ra_guac_autoscale_public_alb.template.cfn.yaml [hunk 5]
```diff
@@ -0,0 +1,623 @@
...
+  MinCapacity:
+    Default: '0'
+    Description: The minimum number of instances for the autoscale group
+    MinLength: '1'
+    Type: String
```
Comment (userhas404d) [inline templates/ra_guac_autoscale_public_alb.template.cfn.yaml]: `Number` type

File: templates/ra_guac_autoscale_public_alb.template.cfn.yaml [hunk 6]
```diff
@@ -0,0 +1,623 @@
...
+    Description: The name (for IAM) or identifier (for ACM) of the SSL certificate to associate with the ALB -- the cert must already exist in the service
+    Type: String
+  SslCertificateService:
+    AllowedValues:
+      - ACM
+      - IAM
+    Default: ACM
+    Description: The service hosting the SSL certificate
+    Type: String
+  URL1:
```
Comment (userhas404d) [inline templates/ra_guac_autoscale_public_alb.template.cfn.yaml]: been a non-issue up to this point but would be nice to have an allowed pattern in here for http/s

File: templates/ra_guac_autoscale_public_alb.template.cfn.yaml [hunk 7]
```diff
@@ -0,0 +1,623 @@
...
+    Default: 'https://accounts.example.com'
+    Description: First custom URL/link to display on the Guac Login page
+    Type: String
+  URL2:
```
Comment (userhas404d) [inline templates/ra_guac_autoscale_public_alb.template.cfn.yaml]: same here with the allowed pattern

File: templates/ra_guac_autoscale_public_alb.template.cfn.yaml (hunk 3 above)
Comment (allez-allez-allez) [reply templates/ra_guac_autoscale_public_alb.template.cfn.yaml]: ok. bumped to t3.micro

File: templates/ra_guac_autoscale_public_alb.template.cfn.yaml (hunk 4 above)
Comment (allez-allez-allez) [reply templates/ra_guac_autoscale_public_alb.template.cfn.yaml]: changed to `Number` type

File: templates/ra_guac_autoscale_public_alb.template.cfn.yaml (hunk 5 above)
Comment (allez-allez-allez) [reply templates/ra_guac_autoscale_public_alb.template.cfn.yaml]: changed to `Number` Type

File: templates/ra_guac_autoscale_public_alb.template.cfn.yaml (hunk 6 above)
Comment (allez-allez-allez) [reply templates/ra_guac_autoscale_public_alb.template.cfn.yaml]: ok. added pattern to check for `https`

File: templates/ra_guac_autoscale_public_alb.template.cfn.yaml (hunk 7 above)
Comment (allez-allez-allez) [reply templates/ra_guac_autoscale_public_alb.template.cfn.yaml]: ok. added pattern to check for `https`

File: templates/ra_guac_autoscale_public_alb.template.cfn.yaml (hunk 2 above)
Comment (allez-allez-allez) [reply templates/ra_guac_autoscale_public_alb.template.cfn.yaml]: updating to "amzn-ami-hvm-2018.03*.2018*-x86_64-gp2". Stack created successfully with the new image search string

File: templates/ra_guac_autoscale_public_alb.template.cfn.yaml [hunk 8]
```diff
@@ -0,0 +1,623 @@
...
+
+    Type: 'AWS::AutoScaling::LaunchConfiguration'
+  Listener:
+    Properties:
+      Certificates:
+        - CertificateArn:
+            !If
+            - UseACM
+            - !Sub 'arn:aws:acm:${AWS::Region}:${AWS::AccountId}:certificate/${SslCertificateName}'
+            - !Sub 'arn:aws:acm:${AWS::Region}:${AWS::AccountId}:server-certificate/${SslCertificateName}'
```
Comment (lorengordon) [inline templates/ra_guac_autoscale_public_alb.template.cfn.yaml]:
```suggestion
            - !Sub 'arn:aws:iam:${AWS::Region}:${AWS::AccountId}:server-certificate/${SslCertificateName}'
```

**Pull request merged**